onnxruntime/core/providers/cpu/nn/conv_transpose_attributes.h

```diff
@@ -176,11 +176,17 @@ struct ConvTransposeAttributes : public ConvAttributes {
 
   void DistributePadding(AutoPadType pad_type, const int64_t& total_pad,
                          int64_t& pad_head, int64_t& pad_tail) const {
+    if (pad_type == AutoPadType::SAME_UPPER || pad_type == AutoPadType::SAME_LOWER) {
+      // TODO(jcwchen): #9740 ORT 1.13 will correct the logic by switching them to meet ONNX spec
+      LOGS_DEFAULT(WARNING) << "The existing bug in the padding distribution for auto_pad type"
```

Comment:
Is it possible to log this only once (during kernel creation)? For a model that is serving high traffic this will have a negative effect on latency.

Reply:
The only way to do this is probably do it when the kernel is constructed since auto_pad will known statically - https://github.com/microsoft/onnxruntime/blob/0702364d7ae7b169670f2ffecc1bc5f13ab15353/onnxruntime/core/providers/cpu/nn/conv_transpose.h#L28. We can't add the logic in ConvTransposeAttributes because that in-turn inherits the attribute parsing from ConvAttributes (and this warning doesn't apply to Conv nodes)

Reply:
Good point. I moved it into ConvTranspose() instead.